packages/flame/test/effects/sequence_effect_test.dart

```diff
@@ -0,0 +1,281 @@
+        ]);
+        expect(effect.controller.duration, 4.5);
+        expect(effect.controller.isRandom, false);
+        expect(effect.controller.completed, false);
+      });
+
+      test('alternating', () {
+        final effect = SequenceEffect(
+          [
+            MoveEffect.to(Vector2(10, 10), EffectController(duration: 3)),
+          ],
+          alternate: true,
+        );
+        expect(effect.controller.duration, 6);
+        expect(effect.controller.isRandom, false);
+      });
+
+      test('infinite', () {
+        final effect = SequenceEffect(
+          [
+            MoveEffect.to(Vector2.zero(), EffectController(duration: 1)),
+          ],
+          alternate: true,
+          infinite: true,
+        );
+        expect(effect.controller.duration, double.infinity);
+        expect(effect.controller.isRandom, false);
+      });
+
+      test('with random effects', () {
+        final randomEffect = MoveEffect.to(
+          Vector2(10, 10),
+          RandomEffectController.uniform(
+            LinearEffectController(0),
+            min: 1,
+            max: 5,
+          ),
+        );
+        final effect = SequenceEffect(
+          [randomEffect],
+          alternate: true,
+          repeatCount: 1000,
+        );
+        expect(
+          effect.controller.duration,
+          closeTo(randomEffect.controller.duration! * 2000, 1e-15),
+        );
+        expect(effect.controller.isRandom, true);
+      });
+
+      test('errors', () {
+        expect(
+          () => SequenceEffect(<Effect>[]),
+          failsAssert('The list of effects cannot be empty'),
+        );
+        expect(
+          () => SequenceEffect(
+            [MoveEffect.to(Vector2.zero(), EffectController(duration: 1))],
+            infinite: true,
+            repeatCount: 10,
+          ),
+          failsAssert(
+            'Parameters infinite and repeatCount cannot be specified '
+            'simultaneously',
+          ),
+        );
+      });
+    });
+
+    group('sequence progression', () {
+      test('simple sequence', () async {
+        final effect = SequenceEffect([
+          MoveEffect.by(Vector2(10, 0), EffectController(duration: 1)),
+          MoveEffect.by(Vector2(0, 10), EffectController(duration: 2)),
+          MoveEffect.by(Vector2(-10, 0), EffectController(duration: 3)),
+          MoveEffect.by(Vector2(30, 30), EffectController(duration: 4)),
+        ]);
+        final component = PositionComponent()..add(effect);
+
+        final game = FlameGame()..onGameResize(Vector2.all(1000));
+        await game.ensureAdd(component);
+        game.update(0);
+
+        // Each point here is spaced 0.1 seconds apart
+        final expectedPositions = [
+          for (var i = 0; i < 10; i++) Vector2(i * 1.0, 0),
+          for (var i = 0; i < 20; i++) Vector2(10, i * 0.5),
+          for (var i = 0; i < 30; i++) Vector2(10 - i / 3, 10),
+          for (var i = 0; i < 40; i++) Vector2(i * 0.75, 10 + i * 0.75),
```

Comment:
I'd use `List.generate` here instead (and on the other places).

Reply:
I tried, and it doesn't look any better or worse than the for-loops:
```dart
        final expectedPositions = <Vector2>[
          ...List.generate(10, (i) => Vector2(i * 1.0, 0)),
          ...List.generate(20, (i) => Vector2(10, i * 0.5)),
          ...List.generate(30, (i) => Vector2(10 - i / 3, 10)),
          ...List.generate(40, (i) => Vector2(i * 0.75, 10 + i * 0.75)),
          Vector2(30, 40),
        ];
```
In other places, though, we need loops running with decrementing counters, so `List.generate` won't work there (or at least it won't be pretty).

Reply:
Looks a bit cleaner I think

Reply:
ok, i've updated the test then